### PR TITLE
Add docker-compose.yaml to connect PostgreSQL db

### PR DESCRIPTION
**Pull request by DominikSerwatka**

### CODE REVIEW

**Comment by jdabrowski** [comment]
Couple notes:

1. we don't want our app to be part of docker-compose.yaml - we just need postgres to be there
2. resolve conflicts before requesting review - you can first create Draft PR (when creating PR there is an option to mark PR as Draft)
3. not sure if we need pgadmin now

**Review by jdabrowski — CHANGES_REQUESTED** [review]
as in comment + comments in code

File: Dockerfile [hunk 1]
```diff
@@ -1,3 +1,4 @@
 FROM eclipse-temurin:17-jdk-alpine
 COPY ./shared-parking-spot-app/target/shared-parking-spot-app-0.0.1-SNAPSHOT.jar shared-parking-spot.jar
+EXPOSE 7777
```
Comment (jdabrowski) [inline Dockerfile]: should not be needed as we don't want to make app part of docker-compose - docker-compose would be a set of tools that in production would be served elsewhere like postgres etc

File: Makefile [hunk 2]
```diff
@@ -1,9 +1,8 @@
 run:
 	cd shared-parking-spot-app/ && \
-    mvn clean install && \
-    cd .. && \
-    docker build -t shared-parking-spot . && \
-    docker run -d -p 7777:7777 shared-parking-spot
+    mvn install -DskipTests=true && \
+    docker-compose up
```
Comment (jdabrowski) [inline Makefile]: not needed

File: docker-compose.yaml [hunk 3]
```diff
@@ -0,0 +1,27 @@
+version: "3.8"
+services:
+  app:
```
Comment (jdabrowski) [inline docker-compose.yaml]: remove

File: docker-compose.yaml [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+      - "7777:7777"
+    depends_on:
+      - postgresqldb
+  postgresqldb:
+    image: postgres
+    environment:
+      - POSTGRES_PASSWORD=postgres
+      - POSTGRES_USER=postgres
+      - POSTGRES_DB=testdb
+  pgadmin:
```
Comment (jdabrowski) [inline docker-compose.yaml]: probably not needed

File: shared-parking-spot-app/pom.xml [hunk 5]
```diff
@@ -29,6 +29,35 @@
 			<artifactId>spring-boot-starter-test</artifactId>
 			<scope>test</scope>
 		</dependency>
+
+		<dependency>
+			<groupId>org.projectlombok</groupId>
+			<artifactId>lombok-maven-plugin</artifactId>
+			<version>1.18.20.0</version>
```
Comment (jdabrowski) [inline shared-parking-spot-app/pom.xml:31]: maven artifacts plugin should be stored in root pom.xml in section "dependencyManagement"

File: shared-parking-spot-app/pom.xml [hunk 6]
```diff
@@ -29,6 +29,35 @@
...
+
+		<dependency>
+			<groupId>org.postgresql</groupId>
+			<artifactId>postgresql</artifactId>
+			<scope>runtime</scope>
+		</dependency>
+
+		<dependency>
+			<groupId>org.springframework.boot</groupId>
+			<artifactId>spring-boot-starter-data-jpa</artifactId>
```
Comment (jdabrowski) [inline shared-parking-spot-app/pom.xml]: the goal of this task was simple to add postgres to our stack - not integrate it
I'm not saying it's wrong but not part of this task

**Review by jdabrowski — CHANGES_REQUESTED** [review]
in comments

**Comment by jdabrowski** [comment]
Let's simply add docker-compose with postgres and proper changes to Makefile

**Comment by DominikSerwatka** [comment]
Ok thanks, I'll try to do it right

**Pull request closed without merging**